tests/contract/test_resource_client.py

```diff
@@ -670,3 +685,28 @@ def test_is_primary_identifier_equal_fail(resource_client):
         {"a": 1, "b": 2, "c": 3},
         {"a": 1, "b": 2, "c": 4},
     )
+
+
+def test_assert_write_only_property_does_not_exist(resource_client):
```

Comment:
what is this testing? a schema with no write only properties? and why are we passing the schema in as a resource model

Reply:
checking for KeyError Exception now.